packages/app/src/cli/models/extensions/specifications/editor_extension_collection.ts

```diff
@@ -0,0 +1,49 @@
+import {BaseSchema} from '../schemas.js'
+import {createExtensionSpecification} from '../specification.js'
+import {zod} from '@shopify/cli-kit/node/schema'
+
+export interface InCollection {
+  handle: string
+}
+
+const InCollectionSchema = zod.object({
+  handle: zod.string(),
+})
+
+export const EditorExtensionCollectionSchema = BaseSchema.extend({
+  in_collection: zod.union([zod.array(InCollectionSchema), zod.array(zod.string())]).optional(),
+  type: zod.literal('editor_extension_collection'),
+})
+
+function isArrayofStrings(arr: string[] | InCollection[]) {
+  return arr.every((item) => typeof item === 'string')
+}
+
+const editorExtensionCollectionSpecification = createExtensionSpecification({
+  identifier: 'editor_extension_collection',
+  schema: EditorExtensionCollectionSchema,
+  appModuleFeatures: (_) => [],
+  deployConfig: async (config, _) => {
+    if (config.in_collection && isArrayofStrings(config.in_collection)) {
+      const inCollection = config.in_collection.map((handle) => {
+        return {
+          handle,
+        }
+      })
+
+      return {
+        name: config.name,
+        handle: config.handle,
+        in_collection: inCollection,
+      }
+    }
+
+    return {
+      name: config.name,
+      handle: config.handle,
+      in_collection: config.in_collection,
+    }
```

Comment:
You can probably simplify this piece of code to avoid duplication and extra functions:

```ts
const in_collection = config.in_collection?.map((item) => {
  return typeof item === 'string' ? {handle: item} : item
})

return {
  name: config.name,
  handle: config.handle,
  in_collection,
}

```